### PR TITLE
Improve README

### PR DESCRIPTION
**Pull request by eoinkelly**
Modernise the README

* have better structure (info for new readers up first). 
* add a table of contents
* document the features of the template more fully and succinctly
* general tidy-up

The easiest way to review this is to click on "View File" on the diff. It will show you the formatted markdown

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -53,21 +53,12 @@ but you may find that many Ackama projects are started with some or all of these
 * `--skip-javascript` skips the setup of JavaScript imports/compilers, Rails 7.0.x is no longer automatically includes [Webpacker](https://github.com/rails/webpacker), instead it uses [Importmap](https://github.com/rails/importmap-rails) by default.
 * `--skip-action-mailbox` skips the setup of ActionMailbox, which you don't need unless you are receiving emails in your application.
 * `--skip-active-storage` skips the setup of ActiveStorage. If you don't need support for file attachments, this can be skipped.
-* `--skip-spring` - you _probably_ want to use this. Spring is great at reducing the start time of Rails processes once
```
Comment (eoinkelly) [inline README.md:56]: not an option in Rails 7

File: README.md [hunk 2]
```diff
@@ -53,21 +53,12 @@ but you may find that many Ackama projects are started with some or all of these
...
-  one is already running, but also tends to cache things more than it should and mask issues (or worse, issues being resolved). We have found that Bootsnap, which is also generated by default takes care of a decent startup time optimisation without the drawbacks of spring.
 * `--skip-action-cable` - if you're not doing things with websockets, you may want to consider skipping this one to avoid
   having an open websocket connection without knowing about it.
-* `--skip-turbolinks` - in a similar vein to spring, Turbolinks is great and providing some user feedback as pages load
```
Comment (eoinkelly) [inline README.md:60]: Not an option in Rails 7

File: README.md [hunk 3]
```diff
@@ -53,21 +53,12 @@ but you may find that many Ackama projects are started with some or all of these
...
-  and speeding up perceived load time, however comes with drawbacks that usually outweight the benefits.
 * `--webpack=react` - this will preconfigure your app to build and serve React code. You only need it if you're going
   to be using React, but adding this during app generation will mean that your codebase supports webpack and React
   components right from the first commit.
 
-If you just want to see what the template does, try running `docker build .` and
-then `docker --rm -it run` the resulting image. You'll be dropped into Bash and
-can explore the generated app in `/apps/template-test`. The image doesn't
-include PostgreSQL right now, so database operations don't work.
```
Comment (eoinkelly) [inline README.md:69]: This info is outdated and a bit confusing.

File: README.md [hunk 4]
```diff
@@ -104,32 +96,6 @@ The template will perform the following steps:
...
-when the application template script is evaluated.
-
-Rails generators are very lightly documented; what you’ll find is that most of
-the heavy lifting is done by [Thor](http://whatisthor.com/). Thor is a tool that
-allows you to easily perform command line utilities. The most common methods
-used by this template are Thor’s `copy_file`, `template`, and `gsub_file`. You
-can dig into the well-organized and well-documented
-[Thor source code](https://github.com/erikhuda/thor) to learn more. If any file
-finishes with `.tt`, Thor considers it to be a template and places it in the
-destination without the extension `.tt`.
```
Comment (eoinkelly) [inline README.md:131]: moved to end of file because it's less important than the usage info

**Review by mischa-s — APPROVED** [review]
😍

**Pull request merged**